File: Cpp/121.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size()<2){
            return 0;
        }
        
        int min_price = prices[0];
        int max_profit = 0;
        for(int i=1;i<prices.size();++i){
            max_profit = max(max_profit, prices[i]-min_price);
            min_price = min(min_price, prices[i]);
        }
        
        return max_profit;
    }
};

int main(){
	vector<int> test={7,1,5,3,6,4};
	Solution s;
	cout<<s.maxProfit(test)<<endl;

	return 0;
}

File: Python3/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        lowest = prices[0]
        for i in range(1, len(prices)):
            if prices[i] < lowest:
                lowest = prices[i]
            else:
                profit = max(profit, prices[i] - lowest)
        return profit
